stdlib/ballerina-config/src/main/java/org/ballerinalang/config/cipher/AESCipherTool.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * @since 0.964.0
+ */
+public class AESCipherTool {
+
+    private static final String ALGORITHM_AES_CBC_PKCS5 = "AES/CBC/PKCS5Padding";
+    private static final String ALGORITHM_AES = "AES";
+    private static final String ALGORITHM_SHA_256 = "SHA-256";
+    private static final String DEFAULT_INITIALIZATION_VECTOR = "0123456789ABCDEF";
+
+    private final Cipher encryptionCipher;
+    private final Cipher decryptionCipher;
+
+    /**
+     * @param userSecret User secret String to encode and decode a value.
+     * @throws AESCipherToolException if Any error occurs when preparing the tool with given user data.
+     */
+    public AESCipherTool(String userSecret) throws AESCipherToolException {
+        this(userSecret, null);
+    }
+
+    /**
+     * @param userSecret User secret String to encode and decode a value.
+     * @param initializingVector Initializing vector to encode and decode values.
+     * @throws AESCipherToolException if Any error occurs when preparing the tool with given user data.
+     */
+    public AESCipherTool(String userSecret, String initializingVector) throws AESCipherToolException {
+        try {
+            IvParameterSpec ivParameterSpec;
+            if (initializingVector == null) {
+                ivParameterSpec = new IvParameterSpec(getSHA256Key(DEFAULT_INITIALIZATION_VECTOR, 16));
+            } else {
+                if (initializingVector.length() > 16) {
+                    throw new InvalidKeyException("Initializing vector can only have 16 byte value");
+                }
+                ivParameterSpec = new IvParameterSpec(getSHA256Key(initializingVector, 16));
+            }
+
+            SecretKey secretKey = new SecretKeySpec(getSHA256Key(userSecret, 32), ALGORITHM_AES);
+            this.encryptionCipher = Cipher.getInstance(ALGORITHM_AES_CBC_PKCS5);
+            this.encryptionCipher.init(Cipher.ENCRYPT_MODE, secretKey, ivParameterSpec);
+            this.decryptionCipher = Cipher.getInstance(ALGORITHM_AES_CBC_PKCS5);
+            this.decryptionCipher.init(Cipher.DECRYPT_MODE, secretKey, ivParameterSpec);
+        } catch (NoSuchPaddingException | NoSuchAlgorithmException | InvalidAlgorithmParameterException
+                | InvalidKeyException e) {
+            throw new AESCipherToolException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * This method is used to encrypt a given value.
+     *
+     * @param value Value to be encrypted.
+     * @return Encrypted value.
+     */
+    public String encrypt(String value) throws AESCipherToolException {
+        try {
+            return encodeBase64(encryptionCipher.doFinal(getBytes(value)));
+        } catch (BadPaddingException | IllegalBlockSizeException err) {
+            throw new AESCipherToolException(err.getMessage(), err);
+        }
+    }
+
+    /**
+     * This method is used to decrypt a given encrypted value.
```

Comment:
This should be encrypted and base64 Encoded string, right ?

Reply:
Done